packages/next/src/server/lib/router-utils/setup-dev.ts

```diff
@@ -993,9 +993,9 @@ async function startWatcher(opts: SetupOpts) {
 
         if (isAppPath) {
           const isRootNotFound = validFileMatcher.isRootNotFound(fileName)
+          hasRootAppNotFound = true
```

Comment:
Do we still need this variable or can we update all the code that referenced it to assume it's always `true`?

Reply:
We still need this as there's 404 page for pages. As even in production we still have this page conditionally, we generate default not-found page only when app page paths existed